systems/overview/12-Glossary.md

```diff
@@ -26,14 +26,34 @@ A component is a software unit with a well-defined interface and explicitly spec
 Extensions are software components that administrators can add (or remove) to their MediaWiki application in order to customize how it looks and works.
 Examples: WikibaseClient, UniversalLanguageSelector, OAuth
 
+## [WikibaseClient](https://www.mediawiki.org/wiki/Extension:Wikibase_Client)
+
+WikibaseClient is a [MediaWiki Extension](#mediawiki-extension) that enables the use and display of Entity data from a [Wikibase Repository](#wikibase-repository) via Lua modules or parser functions. Clients can also use centralized language links.
+
+## Wikibase Repository
+
+Wikibase Repository is a structured data repository providing [Entity](#entity) data. It is implemented as a [MediaWiki extension](#mediawiki-extension) called [WikibaseRepo](https://www.mediawiki.org/wiki/Extension:Wikibase_Repository).
+
 ## Wikibase Extension
 
 Wikibase Extensions are MediaWiki extensions that add further functionality to Wikibase. They depend on the Wikibase extension being installed and directly interact with it.
 Examples: WikibaseLexeme, WikibaseMediaInfo, WikibaseImport
 
+## Entity
+
+Entities are the top level concepts of Wikibase's data model. Items and Properties are the core Entity types of Wikibase. Other types can be added through [Wikibase extensions](#wikibase-extension), such as [Lexemes](https://www.mediawiki.org/wiki/Extension:WikibaseLexeme/Data_Model#Lexeme).
+
+## [Item](https://www.wikidata.org/wiki/Help:Items)
+
+In Wikidata, Items are used to represent all the things in human knowledge, including topics, concepts, and objects. For example, the "1988 Summer Olympics", "love", "Elvis Presley", and "gorilla" are all Items in Wikidata.
+
+## [Property](https://www.wikidata.org/wiki/Help:Properties)
+
+Properties are Entities that describe a relationship between Items (or other Entities) and Values of the property. Typical properties are population (using numbers as values), binomial name (using strings as values), but also has father and author of (both using Items as values).
```

Comment:
1. I spot two lowercase properties here. 
2. Which of the terms that have a Glossary entry should be linked from other Glossary entries? I see Entities, Items and Values here. (wait, Value doesn't even have an entry, although it's capitalised...)
3. I think the property names should be marked somehow, either in quotes or in italic, otherwise the sentence is quite hard to parse.

Reply:
I rebased this as I was about to merge it, then imediatly saw this comment so also tackeled some of this in https://github.com/wmde/wikidata-wikibase-architecture/pull/57/commits/343499dd002d53d6249b3e9f5b8e9a626e176f9a
We should come up with a guide on when we should and shouldn't link to the glossary.

Reply:
Thanks! FWIW this was copied from here https://www.mediawiki.org/wiki/Wikibase/DataModel#Properties